From: Mathieu Trudel-Lapierre <mathieu.trudel-lapierre@canonical.com>
Subject: Don't try to get secrets for ofono modems.

Don't try to get secrets hints and settings name for an ofono connection when
in the prepare stage: it just doesn't apply, since both the APN settings and
modem PIN are handled by ofono directly.

---
 src/devices/wwan/nm-modem-ofono.c |    6 ++++++
 1 file changed, 6 insertions(+)

Index: b/src/devices/wwan/nm-modem-ofono.c
===================================================================
--- a/src/devices/wwan/nm-modem-ofono.c
+++ b/src/devices/wwan/nm-modem-ofono.c
@@ -1001,6 +1001,11 @@ act_stage1_prepare (NMModem *modem,
 	                                      id[2]);
 	g_strfreev (id);
 
+	if (!priv->context_path) {
+		*reason = NM_DEVICE_STATE_REASON_GSM_APN_FAILED;
+		return NM_ACT_STAGE_RETURN_FAILURE;
+	}
+
 	if (priv->connect_properties)
 		g_hash_table_destroy (priv->connect_properties);
 	priv->connect_properties = create_connect_properties (connection);
@@ -1015,6 +1020,7 @@ act_stage1_prepare (NMModem *modem,
 		nm_log_warn (LOGD_MB, "(%s): could not activate context, "
 		             "modem is not registered.",
 		             nm_modem_get_path (modem));
+		*reason = NM_DEVICE_STATE_REASON_MODEM_NO_CARRIER;
 		return NM_ACT_STAGE_RETURN_FAILURE;
 	}
 
